### PR TITLE
Enhanced to evaluate values with custom condition expressions.

### PR DESCRIPTION
**Pull request by spearmin10**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Enhanced to evaluate values with custom condition expressions.

The capability has initially been requested as a new script `If-Then-Else-Ex`.
However it has been decided to update it in the pull request (https://github.com/demisto/content/pull/11909), then I've enhanced the current `If-Then-Else`.

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by dantavori** [comment]
@spearmin10 please reach me on Slack (Dan Tavori) to schedule a quick call about your enhancement.

**Comment by dantavori** [comment]
Had a conversation with @spearmin10 and we decided to keep the current PR with the enhancement. In addition, we decided to add the lhs, op, rhs arguments as implemented in the original PR (https://github.com/demisto/content/pull/11909).

**Review by spearmin10 — COMMENTED** [review]
Reverted 1_3_36.md

**Comment by spearmin10** [comment]
@dantavori Please review the update.

**Comment by spearmin10** [comment]
@dantavori I've updated, please review again.

File: Packs/CommonScripts/ReleaseNotes/1_3_41.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### If-Then-Else
+  - Enhanced to evaluate values with custom condition expressions
```
Comment (dantavori) [inline Packs/CommonScripts/ReleaseNotes/1_3_41.md:4]:
```suggestion
  - Enhanced to evaluate values with custom condition expressions.
```

**Review by dantavori — CHANGES_REQUESTED** [review]
Looks good, we are close to finish. See my small notes, make the last fixes and resolve the conflicts (update the pack version again after you pull from the base branch). We can merge it tomorrow.
Regarding the renaming of `condition` argument, I meant to change it to `op` only if its pre-defined values are the optional operators (and not the whole conditions), but let's keep them the way they are and change it back to `condition`. Sorry for the misunderstanding.

File: Packs/CommonScripts/Scripts/IfThenElse/IfThenElse.js [hunk 2]
```diff
@@ -1,6 +1,141 @@
...
+function getOperator(lhs_name, rhs_name, condition) {
+    var operator = condition.substr(
+        lhs_name.length,
+        condition.length - lhs_name.length - rhs_name.length);
+
+    const specials = [" ", "!", "=", "~", "<", ">"];
+    if (specials.indexOf(operator.slice(0, 1)) < 0 ||
+        specials.indexOf(operator.slice(-1)) < 0 ){
+        throw `Invalid condition: ${condition}`;
+    }
```
[the remark this answers is not in the document]
Comment (spearmin10) [reply Packs/CommonScripts/Scripts/IfThenElse/IfThenElse.js:51]: The check is evaluated one time for a condition and an exception is also raised one time for it.

File: Packs/CommonScripts/Scripts/IfThenElse/IfThenElse.yml [hunk 3]
```diff
@@ -11,20 +11,51 @@ comment: A transformer for simple if-then-else logic.
...
+  - value in list rhs
+  - value not in list rhs
+  description: The condition expression to evaluate such as lhs==rhs or lhs!=rhs.
+    It evaluate as value==equals if this is empty for backward compatibility.
+- name: lhs
+  description: The value to compare in the condition (left hand side)
+- name: rhs
+  description: The value to compare in the condition (right hand side)
+- name: options
+  description: 'Option flags (comma separated list): available options are `case_insensitive`,
```
[the remark this answers is not in the document]
Comment (spearmin10) [reply Packs/CommonScripts/Scripts/IfThenElse/IfThenElse.yml:53]: Available options are listed in the description as below. Could you let me know what the "valid types" means?

Option flags (comma separated list): available options are `case_insensitive`, `regex_dot_all`, `regex_multiline`, `regex_full_match`, `input_data_type:then=<type>`, `input_data_type:else=<type>`, `input_data_type:lhs=<type>` and `input_data_type:rhs=<type>`.

[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/CommonScripts/Scripts/IfThenElse/IfThenElse.yml:53]: I meant, write in the description of the argument what are the types: json, value or raw.

File: Packs/CommonScripts/ReleaseNotes/1_3_41.md [hunk 4]
```diff
@@ -1,4 +0,0 @@
-
```
Comment (roysagi) [inline Packs/CommonScripts/ReleaseNotes/1_3_41.md:1]: Please revert this change - we do not delete existing release notes files

**Comment by spearmin10** [comment]
@dantavori  @roysagi I've updated the description to your review comments.

**Review by dantavori — APPROVED** [review]

**Pull request merged**